tdvt/tdvt/config_gen/test_creator.py

```diff
@@ -0,0 +1,163 @@
+from ..resources import get_root_dir
+
+
+class TestCreator:
+
+    def __init__(self, json_file, datasource_name, output_dir=os.getcwd()):
+        self.json_file: Path = Path(json_file)
+        self.datasource_name: str = datasource_name
+        self.output_dir: Path = Path(output_dir)
+
+    def _json_to_header_list(self) -> List[str]:
+        json_file = self.json_file
+        with open(json_file, 'r') as f:
+            json_dict = json.load(f)
+        headers = list(json_dict.keys())
+        return headers
+
+    def parse_json_to_list_of_columns(self) -> List[str]:
+        if not self._check_output_dir_and_input_json_exist():
+            print("{} does not exist.".format(self.json_file))
+            sys.exit(1)
+
+        headers = self._json_to_header_list()
+
+        return headers
+
+    def write_test_files(self, list_to_write: List[str]) -> None:
+        output_affix = 'setup.'
+        for item in list_to_write:
+            output_file_name = output_affix + self.datasource_name + '.' + item + '_column.txt'
+            output_path = self.output_dir / output_file_name
+            with open(output_path, 'w') as out:
+                print('writing to {}'.format(output_path))
+                logging.info("writing to {}".format(output_path))
+                out.write(item + '\n')
+            logging.info("Successfully wrote to {}".format(output_path))
+
+    def _check_output_dir_and_input_json_exist(self) -> bool:
+        result = True
+        if self.json_file.is_file() and self.output_dir.is_dir():
+            logging.info("Source JSON file found.")
+            logging.info("Output directory found.")
+        else:
+            if not self.json_file.is_file():
+                logging.error("JSON file does not exist at indicated path: {}.".format(self.json_file))
+            if not self.output_dir.is_dir():
+                logging.error("Output directory {} does not exist.".format(self.output_dir))
+            result = False
+
+        return result
+
+    @staticmethod
+    def _parse_col_constants() -> Dict[str, List[str]]:
+        """
+        Parses TEST_ARGUMENT_DATA_TYPES from constants file to get the column names.
+        """
+
+        col_type_map = {
+            k: [] for k in TEST_ARGUMENT_DATA_TYPES.keys()
+        }
+        return col_type_map
+
+    def _map_user_cols_to_test_cols(self) -> Dict[str, Dict[str, str]]:
+        with open(self.json_file, 'r') as f:
+            user_cols_dict = json.load(f)
+        return user_cols_dict
+
+    def map_user_cols_to_test_args(self) -> Dict[str, List[str]]:
+        user_cols_dict = self._map_user_cols_to_test_cols()
+        test_args_dict = self._parse_col_constants()
+        for user_col in user_cols_dict.keys():
+            for test_col in TEST_ARGUMENT_DATA_TYPES.keys():
+                if user_cols_dict[user_col] == TEST_ARGUMENT_DATA_TYPES[test_col]:
```

Comment:
Should we allow matching on name as well?

Reply:
Good idea, I'll add it.

Reply:
On reflection, I'm thinking this should be a separate workflow; if you have 1:1 mapping, we should allow you to give us a json object of the mapping, saving the user from all the other typing. It could also allow them to run all the tests, instead of a subset of the tests like this PR enables. I'd prefer to add that in a small follow-on PR I think.